src/traces/histogram/calc.js

```diff
@@ -102,12 +102,20 @@ function calc(gd, trace) {
         };
     }
 
+    if(!gd._fullLayout._roundFnOpts) gd._fullLayout._roundFnOpts = {};
+    var groupName = trace['_' + mainData + 'bingroup'];
+    var roundFnOpts = {leftGap: Infinity, rightGap: Infinity};
+    if(groupName) {
+        if(!gd._fullLayout._roundFnOpts[groupName]) gd._fullLayout._roundFnOpts[groupName] = roundFnOpts;
+        roundFnOpts = gd._fullLayout._roundFnOpts[groupName];
```

Comment:
Non-blocking, 
We could rewrite this:
```js
if(!gd._fullLayout._roundFnOpts[groupName]) {
  gd._fullLayout._roundFnOpts[groupName] = roundFnOpts;
}
roundFnOpts = gd._fullLayout._roundFnOpts[groupName];
```
to make it easier to read 3 `gd._fullLayout._roundFnOpts[groupName]`.